### PR TITLE
SIWW auth

### PR DESCRIPTION
**Pull request by r-czajkowski**
This PR adds Sign-In With Wallet auth to the Acre dapp. When a user connects the wallet we create a session by sending a request to the Acre API backend. The session is valid for 3 hours.

### SIWW flow:
1. Dapp asks backend for the session.
2. If the session exists and it matches the current connected address, a user is logged in.
3. If a session exists but does not match the current connected address the dapp deletes the session and asks backend for a new nonce (session id).
4. The user must sign the SIWW message with a given nonce.
5. Dapp sends the signature, message, and public key to the backend to verify the signature.
6. If the message is valid, the backend returns the session id in cookies and the user is logged in.

### CODE REVIEW

**Review by kkosiorowska — COMMENTED** [review]
Not tested yet, I left a few minor comments.

File: dapp/src/utils/orangeKit.ts [hunk 1]
```diff
@@ -47,6 +51,10 @@ const createSignInWithWalletMessage = (address: string) => {
     issuedAt: new Date().toISOString(),
     version: "1",
     networkFamily: "bitcoin",
+    expirationTime: new Date(
+      Date.now() + ACRE_SESSION_EXPIRATION_TIME,
+    ).toISOString(),
+    nonce,
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/utils/orangeKit.ts]: c0644d6

File: dapp/src/utils/index.ts [hunk 2]
```diff
@@ -17,3 +17,4 @@ export { default as userAgent } from "./userAgent"
 export { default as referralProgram } from "./referralProgram"
 export { default as mezoPortalAPI } from "./mezoPortalApi"
 export { default as router } from "./router"
+export * from "./acre-api"
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/utils/index.ts]: b4fbbd0

File: dapp/src/constants/time.ts [hunk 3]
```diff
@@ -10,3 +10,6 @@ export const REFETCH_INTERVAL_IN_MILLISECONDS =
   ONE_SEC_IN_MILLISECONDS * ONE_MINUTE_IN_SECONDS * 5
 
 export const DATE_FORMAT_LOCALE_TAG = "us-US"
+
+// 3 hours
+export const ACRE_SESSION_EXPIRATION_TIME = 10800000
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/constants/time.ts]: 75fbc17

File: dapp/src/components/ConnectWalletModal/ConnectWalletButton.tsx [hunk 4]
```diff
@@ -84,16 +91,45 @@ export default function ConnectWalletButton({
...
+          btcAddress,
+          session.nonce,
+        )
+
+        const signedMessage = await signMessageAsync({
+          message,
+          connector: orangeKit.typeConversionToConnector(connectedConnector),
+        })
+
+        await createSession(message, signedMessage)
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/ConnectWalletModal/ConnectWalletButton.tsx]: 9bb6bf4

**Review by kkosiorowska — DISMISSED** [review]
LGTM, I tested the solution and it seems to work well. 🚀🚀🚀



Screenshot 2024-09-13 at 10 30 57

**Review by kkosiorowska — APPROVED** [review]
I tested deploy previews and it seems to work well. 🚀

Screenshot 2024-09-18 at 10 03 25

Screenshot 2024-09-18 at 10 02 32

**Pull request merged**